### ISSUE TITLE
[regression] npm can not find package.json in git repository

### ISSUE THREAD

**Issue by dotnetCarpenter** [npm/npm#19409]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

npm can not find package.json in git repository

#### How can the CLI team reproduce the problem?

1. `npm i -g git@github.com:dotnetCarpenter/node-ecstatic.git`

Gist of log: https://gist.github.com/dotnetCarpenter/2f0c5c7ebb38b0db331ed20489562cba

### supporting information:

 - `npm -v` prints: 5.6.0
 - `node -v` prints: v8.9.3
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Linux
 - Network issues:
   - Geographic location where npm was run: Denmark
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by dotnetCarpenter** [comment]
With yarn I get:

```
yarn global add  git@github.com:dotnetCarpenter/node-ecstatic.git
yarn global v1.3.2
[1/4] Resolving packages...
error Command failed.
Exit code: 128
Command: git
Arguments: ls-remote --tags --heads github.com:dotnetcarpenter/node-ecstatic.git
Directory: /home/dotnet/projects/napp/loader
Output:
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

This might be an issue with a change on github.com?

Or it might be related to #16854

**Comment by dotnetCarpenter** [comment]
OK, this just seems very wrong:
```
7 silly fetchPackageMetaData error for git@file:github.com:dotnetCarpenter/node-ecstatic.git Could not install from "github.com:dotnetCarpenter/node-ecstatic.git" as it does not contain a package.json file.
8 verbose stack Error: ENOENT: no such file or directory, open '/home/dotnet/projects/napp/loader/github.com:dotnetCarpenter/node-ecstatic.git/package.json'
```

**Comment by vdhpieter** [comment]
I have the same issue (with another package) where you able to resolve it?

**Comment by vdhpieter** [comment]
Just tried this with npm v5.5.1 and it seems to work for me, so probably a bug in 5.6.0

**Comment by dotnetCarpenter** [comment]
@vdhpieter No. I haven't resolved it. I instead installed from my local folder. The work around was fine because I was just testing a forked project but not sustainable solution. Most use-cases will be in distributed `package.json` but this is currently broken. Since it works in 5.5.1, it's a **regression**. *Bad bad npm*
